pygments/lexers/vyper.py

```diff
@@ -0,0 +1,129 @@
+"""
+    pygments.lexers.vyper
+    ~~~~~~~~~~~~~~~~~~~~~
+    
+    Lexer for the Vyper Smart Contract language.
+
+    :copyright: Copyright 2006-2023 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, bygroups
+from pygments.token import (Comment, String, Name, Keyword, Number,
+                            Operator, Punctuation, Text, Error)
+
+class VyperLexer(RegexLexer):
+    name = 'Vyper'
+    aliases = ['vyper']
+    filenames = ['*.vy']
+    url = "https://vyper.readthedocs.io/"
+
+
+    tokens = {
+        'root': [
+            # Whitespace
+            (r'\s+', Text.Whitespace),
+
+            # Comments
+            (r'#.*$', Comment.Single),
+            (r'\"\"\"', Comment.Multiline, 'multiline-comment'),
+
+            # Strings
+            (r"'([^'\\]*(?:\\.[^'\\]*)*)'", String.Single),
+            (r'"([^"\\]*(?:\\.[^"\\]*)*)"', String.Double),
+
+            # Numeric Literals
+            (r'\b0x[0-9a-fA-F]+\b', Number.Hex),
+            (r'\b\d{1,3}(?:_\d{3})*\b', Number.Integer),
+            (r'\b\d+\.\d*\b', Number.Float),
+
+            # Keywords
+            (r'\b(def|event|pass|return|for|while|if|elif|else|assert|raise|import|in|struct|implements|interface|from)\b', Keyword),
```

Comment:
Use the `words` function for this which creates an optimized regex (and the other cases below as well.)